### ISSUE TITLE
PushButton not keyboard accessible

### ISSUE THREAD

**Issue by jacob-willden** [macosui/macos_ui#575]
**Description**

Every interactive element in an app (including buttons, form fields, and hyperlinks) [should be able to focused](https://webaim.org/techniques/keyboard/) using the <kbd>Tab</kbd> key. While many of macos_ui's elements are keyboard focus-able, `PushButton` is not, meaning it cannot be activated by desktop users who don't have a mouse, or by users using other assistive technologies. 

Any apps that use `PushButton`, as it currently stands, will fail [Success Criteria 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard.html) for the Web Content Accessibility Guidelines (which also applies to native apps), which could also result in digital accessibility lawsuits for any entities that provide such apps.

**Steps To Reproduce**

1. Make sure the "Keyboard navigation" switch in your MacOS System Settings is turned on (in the "Keyboard" section of the settings).
2. Make a new Flutter project by running the command `flutter create my_project`.
3. Run `flutter pub add macos_ui` to add macos_ui to the project.
4. Replace the code in the `main.dart` file in the `lib` sub-directory with the code below.
5. Run the program, either for Web by running `flutter run -d chrome` or for MacOS by running `flutter run` and selecting the "macOS (macos)" option (oddly, `flutter run macos` does not work).
6. While in the currently running app, press the <kbd>Tab</kbd> key a few times and note that only the "Email" field can be focused.

<details>
<summary>main.dart</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:macos_ui/macos_ui.dart';

void main() {
  runApp(const EmailSubscribeApp());
}

class EmailSubscribeApp extends StatelessWidget {
  const EmailSubscribeApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MacosApp(
      title: 'Unsolicited emails drive our business!',
      home: EmailSubscribePage()
    );
  }
}

class EmailSubscribePage extends StatefulWidget {
  const EmailSubscribePage({super.key});

  @override
  EmailSubscribePageState createState() => EmailSubscribePageState();
}

class EmailSubscribePageState extends State<EmailSubscribePage> {
  @override
  Widget build(BuildContext context) {
    return MacosWindow(
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Label(text: Text('Email')),
            Padding(
              padding: EdgeInsets.only(bottom: 8.0),
              child: MacosTextField(
                placeholder: 'johndoe@example.com',
                placeholderStyle: TextStyle(color: Color.fromARGB(0xFF, 0x75, 0x75, 0x75)),
                style: TextStyle(color: Color.fromARGB(0xFF, 0x22, 0x22, 0x22)),
              )
            ),
            PushButton(
              controlSize: ControlSize.large,
              semanticLabel: 'Subscribe to spam!',
              child: Text('Subscribe to spam!'),
              onPressed: () async {}
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

**Expected behavior**

I expected to be able to focus the submit button (like I can for the "Email" text field) and activate it, but I cannot.

**Screenshots**

Running the app for MacOS native and hitting <kbd>Tab</kbd> to focus the "Email" field, then hitting <kbd>Tab</kbd> again, which keeps the focus on the "Email" field because there are no other focus-able elements the app:
<img width="360" alt="Focus stuck on the Email field in MacOS native." src="https://github.com/user-attachments/assets/0c9992bf-b6f8-4667-991c-a13adf16c65c" />

Running the app for Web and hitting <kbd>Tab</kbd> to focus the "Email" field:
<img width="593" alt="Focus on the Email field in Web." src="https://github.com/user-attachments/assets/41f6bcc4-194c-4af6-a816-49b510486ce8" />

Then hitting <kbd>Tab</kbd> again, which skips over the submit button:
<img width="593" alt="Focus outside the window in Web." src="https://github.com/user-attachments/assets/198a0ccf-5338-46b0-b5a5-8663b879a8c9" />

**Logs**

<details>
  <summary>Logs</summary>

The log output from running `flutter run --verbose` exceeds GitHub's character limit for issues, so I'm linking to it [in a paste-bin](https://gist.github.com/jacob-willden/e4928f2b11763ed670e2fa35b5960ee7). I could not find any errors in the log that appeared relevant to this issue.

Output of `flutter analyze`:
```
Analyzing pushbutton_example...                                                
No issues found! (ran in 0.7s)
```

Output of `flutter doctor -v`:
```
[✓] Flutter (Channel stable, 3.32.4, on macOS 15.5 24F74 darwin-arm64, locale en-US) [174ms]
    • Flutter version 3.32.4 on channel stable at /Users/jacob/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6fba2447e9 (2 weeks ago), 2025-06-12 19:03:56 -0700
    • Engine revision 8cd19e509d
    • Dart version 3.8.1
    • DevTools version 2.45.1

[✗] Android toolchain - develop for Android devices [134ms]
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 16.4) [544ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16F6
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [7ms]
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[!] Android Studio (not installed) [6ms]
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).

[✓] Connected device (2 available) [5.6s]
    • macOS (desktop) • macos  • darwin-arm64   • macOS 15.5 24F74 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 138.1.80.113

[✓] Network resources [601ms]
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

</details>

**Comment by Adrian-Samoticha** [comment]
If I understand correctly, the push button should only be focusable when the Keyboard Navigation is enabled in the system’s settings. Unfortunately, that makes the fix significantly more involved, and (likely) impossible on the web, since you cannot access macOS’ system settings on the web (as far as I am aware, at least).

**Comment by jacob-willden** [comment]
True, `macos_ui` only officially supports MacOS. I was using the Web version in addition to MacOS to illustrate that the button cannot be focused in either of them. Sorry, I was unclear about that.

Would it be possible to make `PushButton`s keyboard focus-able on just MacOS then?

**Comment by Adrian-Samoticha** [comment]
Probably, but—again—it’d be somewhat difficult, as it would require the development of a native plugin to retrieve the system settings.

Honestly, accessibility is a topic that I’ve been kind of pushing ahead of me due to its complexity (macOS has a large number of accessibility settings which we would all need to respect) and my general lack of time that I can dedicate to the project. It’s planned, there’s absolutely no timeline as to when you can expect it.

**Comment by jacob-willden** [comment]
Thanks for getting back to me. Sorry that MacOS's native APIs have made this difficult. My organization has found another framework that we can use, so we're fine with that. Best of luck!

**Comment by jacob-willden** [comment]
Update: I discovered that at least when building for MacOS, the issue of the button being inaccessible is not exclusive to this framework, but is also present in Material. Sorry for wrongfully jumping to the conclusion that this was exclusive to `macos_ui`!
